### PR TITLE
Fix/add listing shortcode parameter

### PR DESCRIPTION
**Pull request by Armanul46**
https://app.clickup.com/t/1tth26k

### CODE REVIEW

**Review by kowsar89 — CHANGES_REQUESTED** [review]
If default listing type is different than current listing type, it saves things as default listing type

**Comment by obiPlabon** [comment]
The form is displayed properly when the slug is used and a directory exists with that slug. Fatal error otherwise

```
Fatal error: Uncaught Error: Class 'Helper' not found in /Users/macbookpro/Local Sites/testdirectorist/app/public/wp-content/plugins/directorist/templates/listing-form/add-listing-notype.php on line 12 ( ! ) Error: Class 'Helper' not found in /Users/macbookpro/Local Sites/testdirectorist/app/public/wp-content/plugins/directorist/templates/listing-form/add-listing-notype.php on line 12 Call Stack #TimeMemoryFunctionLocation 10.0003409184{main}( ).../index.php:0 20.0003409456require( '/Users/macbookpro/Local Sites/testdirectorist/app/public/wp-blog-header.php' ).../index.php:17 30.12885096736require_once( '/Users/macbookpro/Local Sites/testdirectorist/app/public/wp-includes/template-loader.php' ).../wp-blog-header.php:19 40.13675134776include( '/Users/macbookpro/Local Sites/testdirectorist/app/public/wp-content/themes/twentytwentyone/page.php' ).../template-loader.php:106 50.21425568712get_template_part( ).../page.php:17 60.21435569232locate_template( ).../general-template.php:204 70.21435569392load_template( ).../template.php:716 80.21435569776require( '/Users/macbookpro/Local Sites/testdirectorist/app/public/wp-content/themes/twentytwentyone/template-parts/content/content-page.php' ).../template.php:772 90.21565570608the_content( ).../content-page.php:29 100.21675570608apply_filters( ).../post-template.php:253 110.21675571008WP_Hook->apply_filters( ).../plugin.php:189 120.21725639704do_shortcode( ).../class-wp-hook.php:303 130.21725643960preg_replace_callback ( ).../shortcodes.php:228 140.21725644640do_shortcode_tag( ).../shortcodes.php:228 150.21735645112Directorist\ATBDP_Shortcode->add_listing( ).../shortcodes.php:356 160.21855646480Directorist\Directorist_Listing_Form->render_shortcode( ).../class-shortcode.php:235 170.22165659192Directorist\Helper::get_template_contents( ).../ListingForm.php:865 180.22165675704Directorist\Helper::get_template( ).../trait-uri-helper.php:14 190.22175676144include( '/Users/macbookpro/Local Sites/testdirectorist/app/public/wp-content/plugins/directorist/templates/listing-form/add-listing-notype.php' ).../trait-uri-helper.php:73
```

**Review by kowsar89 — DISMISSED** [review]
works fine now

**Review by obiPlabon — CHANGES_REQUESTED** [review]
Please fix the fatal error issue.

**Comment by Armanul46** [comment]
Fixed the undefined class issue

**Review by kowsar89 — APPROVED** [review]

**Comment by obiPlabon** [comment]
@Armanul46 `directorist-alert directorist-alert-info` please use these class and style the error text.

**Review by obiPlabon — APPROVED** [review]
Looks good

File: includes/model/ListingForm.php [hunk 1]
```diff
@@ -860,10 +867,21 @@ public function render_shortcode( $atts ) {
 		} else {
 			// if no listing type exists
 			if ( $listing_type_count == 0 ) {
-				return Helper::get_template_contents( 'listing-form/add-listing-notype' );
+
+				if( ! directory_types() ) {
+					if( current_user_can('manage_options') || current_user_can('edit_pages') ) {
+						$args['error_notice'] = sprintf( __('Please add a directory type first %s', 'directorist' ), '<span><a href="'. admin_url() .'/edit.php?post_type=at_biz_dir&page=atbdp-directory-types">Add Now</a></span>' );
```
Comment (kowsar89) [inline includes/model/ListingForm.php:873]: 1. Why extra <span>?
2. Double slash in link

File: includes/model/ListingForm.php [hunk 2]
```diff
@@ -860,10 +867,21 @@ public function render_shortcode( $atts ) {
...
+					} else {
+						$args['error_notice'] = __('There\'s something unexpected happen. Please contact site admin.', 'directorist');
+					}
+				} else {
+					$args['error_notice'] = __('Notice: your given directory type is not valid. Please use a valid directory type', 'directorist');
```
Comment (kowsar89) [inline includes/model/ListingForm.php:878]: 'your' - should be capital letter

**Review by kowsar89 — CHANGES_REQUESTED** [review]

**Review by kowsar89 — APPROVED** [review]
seems ok

**Pull request merged**